File: programmers/lv3/apart_station.cpp
#include <vector>
using namespace std;

int add_giji(int cnt, int len) {
    if (cnt <= 0) return 0;
    return cnt / len + (cnt % len != 0);
}

int solution(int n, vector<int> stations, int w)
{
    int ans = 0;
    int len = w * 2 + 1;
    int start = 1;
    
    for (int s : stations) {
        ans += add_giji(s - w - start, len);
        start = s + w + 1;
    }
    ans += add_giji(n - start + 1, len);
    return ans;
}

File: programmers/lv3/apart_station.swift
import Foundation

func solution(_ n:Int, _ stations:[Int], _ w:Int) -> Int{
    var answer = 0
    var start = 1
    let len = w * 2 + 1
    
    func add(_ range: Int) -> Int {
        if range < 0 {
            return 0
        }
        return range / len + (range % len == 0 ? 0 : 1)
    }
    
    for s in stations {
        answer += add(s - w - start)
        start = s + w + 1
    }
    answer += add(n + 1 - start)
    return answer
}
